play/src/front/Chat/Components/Avatar.svelte

```diff
@@ -0,0 +1,86 @@
+<script lang="ts">
+    import WokaFromUserId from "../../Components/Woka/WokaFromUserId.svelte";
+
+    export let avatarUrl: string | null = null;
+    export let userId: number | string | null = null;
+    export let fallbackFirstLetter = "A";
+    export let color: string | null = null;
+    export let isChatAvatar = false;
+
+    const colors = [
+        "CB4288",
+        "C5458A",
+        "BF488C",
+        "B94C8D",
+        "B34F8F",
+        "AD5291",
+        "A75593",
+        "A15995",
+        "9B5C96",
+        "955F98",
+        "8F629A",
+        "89669C",
+        "83699E",
+        "7D6C9F",
+        "776FA1",
+        "7173A3",
+        "6B76A5",
+        "6579A7",
+        "5F7CA8",
+        "5980AA",
+        "5383AC",
+        "4D86AE",
+        "4789B0",
+        "418DB1",
+        "3B90B3",
+        "3593B5",
+    ];
+
+    function getColorByFirstLetter(firstLetter: string) {
+        if (!firstLetter) {
+            return "black";
+        }
+        const indexColor = firstLetter.toLowerCase().charCodeAt(0) - 97;
+        return colors[indexColor];
```

Comment:
We need a fallback here if the first letter is not... a letter.
For instance if this is a symbol or an emoji.


You should instead use 
getColorByString() from "colors.ts" that already does this.

Or take a look at this one that does in addition keep the intensity level using "hue"

https://github.com/thecodingmachine/workadventure/blob/8354cf1ff85ad7965cb9f8c3a7a44360026aae12/play/src/front/Stores/FollowStore.ts#L79-L81

Reply:
An issue was opened here: https://github.com/workadventure/workadventure/issues/4034